processor/otel/traces.go

```diff
@@ -688,6 +702,7 @@ func TranslateSpan(spanKind pdata.SpanKind, attributes pdata.AttributeMap, event
 		}
 	}
 
+	serviceTarget.Name = peerService
```

Comment:
@SylvainJuge @axw For `service.target.name` I have kept `peer.service` as highest priority unlike `destination.service.resource` which has `peer.address` as highest priority followed by `peer.service`.

The main motivation for this was that `peer.service` aligns well the definition of `service.target.name`. Additionally IIUC, `peer.address` is defined in the deprecated opentracing spec. The somewhat equivalent of `peer.address` in open-telemetry spec is `net.peer.*` which is handled in a different way.

Maybe it is time to remove handing `peer.address` entirely? What are your thoughts on this?

Reply:
I think those priorities make sense.

> Maybe it is time to remove handing peer.address entirely? What are your thoughts on this?

IMO we should keep it for now. I think we should have a broader discussion about deprecating both the Jaeger intake and OpenTracing bridges at some point.

Reply:
After checking the usage of `peer.service` within the OpenTelemetry Java agent:
- when there is no value for `peer.service`, the value of `peer.service` is set t the same value as `peer.ip`
- the value of `peer.service` contains the host name for HTTP spans (or whatever is in URL).
- for some instrumentations like gRPC and Dubbo the service name is not set to a protocol-specific name (gRPC has a concept of service name which is not used here).

With that, I think the best option (for now) would be to :
- make the `peer.service` value only override the hostname part of the resource when applicable (HTTP spans and RPC spans)
- ignore the value of `peer.service` for the database spans as the values will contain the host/IP of the database, whereas we use it for the database name.

Reply:
@SylvainJuge thanks for the additional context. That's kind of annoying, but reading the semantic conventions again, it's only a "SHOULD", so I suppose you're right that we should give `peer.service` higher precedence in general.

> - make the peer.service value only override the hostname part of the resource when applicable (HTTP spans and RPC spans)
> - ignore the value of peer.service for the database spans as the values will contain the host/IP of the database, whereas we use it for the database name.

This sounds a little bit complicated. Can we simplify it, and only use `peer.service` for any span other than HTTP, RPC, database, and messaging? i.e. just as a fallback. WDYT?

Reply:
Yes, that would definitely be a fair compromise here, let's just make `peer.service` have higher priority on the `service.target.name` field and call it a day. From the user perspective that would also be a bit better as the value set explicitly in the code would be reflected in the map and dependencies.

Reply:
@SylvainJuge @axw  Thanks for the review. IIUC, the conclusion here is to use `peer.service` to have highest priority when inferring `service.target.name` for **ALL** span types. Let me know if I am misunderstanding above discussion.

Reply:
Sorry, crossed wires between me and @SylvainJuge I think. I think that would be ideal _if_ the OTel SDK didn't default to using `peer.ip`.

@SylvainJuge what I meant was:

 - for HTTP, RPC, database, and messaging spans: follow the rules laid out in https://github.com/elastic/apm-server/issues/7980
 - for all other span types, use `peer.service` for `service.target.name`

Reply:
Thanks for the clarification @axw, using `peer.service` only on "other" span types makes sense and allows to keep maximum compatibility here, it should even allow to prevent duplication of dependencies with a mixed set of OTel and our agents.

For `peer.ip` I don't have any strong opinion, thus I'll rely on you for this one.

Reply:
Thanks for the clarification. I have updated the PR and the tests in the description as per our conclusion.